---
title: "GSE75748 data analysis"
author: "German Novakovskiy"
date: '4 March 2018'
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Here we analyze gene expression data from [this paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1033-x). Some analysis steps are made according to [this workflow article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/).

```{r}
#source("https://bioconductor.org/biocLite.R")
#biocLite("GEOquery", "biomaRt")
library(knitr)
suppressWarnings(suppressMessages(library(GEOquery)))
suppressWarnings(suppressMessages(library(biomaRt)))
#install.packages("tidyverse")
suppressWarnings(suppressMessages(library(tidyverse)))
#install.packages("data.table")
suppressWarnings(suppressMessages(library(data.table)))
#install.packages("reshape2")
suppressWarnings(suppressMessages(library(reshape2)))
suppressWarnings(suppressMessages(library(dplyr)))
suppressWarnings(suppressMessages(library(ggplot2)))
suppressWarnings(suppressMessages(library(cowplot)))
suppressWarnings(suppressMessages(library(limma)))
suppressWarnings(suppressMessages(library(tibble)))
```

Let's read our data into R (GSE75748):
```{r}
if (file.exists("GSE75748.Rdata")) {
    # if previously downloaded
    load("GSE75748.Rdata")
} else {
    # Get geo object that contains our data and phenotype information
    geo_obj <- getGEO("GSE75748", GSEMatrix = TRUE)
    #geo_obj contains 1810 entries ((1018 + 758) single cells + (19 + 15) bulk RNA-seq)
    geo_obj <- geo_obj[[1]]
    save(geo_obj, file = "GSE75748.Rdata")
}
```

Further analysis of the downloaded data:
```{r}
#show data structure
show(geo_obj)

```
Let's load metadata:
```{r}
geo_metadata <- pData(geo_obj)

#due to complexity of metadata table it's more convenient to create your own (see below)
```

assayData has 0 features, it's not possible to load expression data from this dataset. That's why we will use expression data downloaded directly from the website
```{r}
#cell type
bulk_cell_type_ec <- read.csv("GSE75748_bulk_cell_type_ec.csv/GSE75748_bulk_cell_type_ec.csv", header=TRUE)
#time course
bulk_time_course_ec <- read.csv("GSE75748_bulk_time_course_ec.csv/GSE75748_bulk_time_course_ec.csv", header=TRUE)
```

Let's create metadata data frame:
```{r}
sample_names <- colnames(bulk_time_course_ec[-1]) #without gene name
time_fact <- paste(c(12,24,36,72,96), sep="", 'h') #factors of time
time_fact <- rep(time_fact, each=3)

time_int <- rep(c(12,24,36,72,96), each=3) #time as continious variable

metadata <- data.frame(samples = sample_names, time = time_fact, age = time_int)

metadata
```
```{r}
str(metadata)
```

Our primary goal here is to analyze bulk_time_course data (each stage with the next stage, step-by-step).

Now let's do some sanity check: let's see how gene expression is distributed across different samples in different time points:

```{r}
#change first column name to gene
colnames(bulk_time_course_ec)[1] <- "gene"
meltedBultTimeCourseEc <- melt(bulk_time_course_ec, id='gene')

meltedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```
Let's look at density plots:
```{r}
meltedBultTimeCourseEc %>% 
  ggplot(aes(x = value, color = variable)) +
  geom_density() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

We can look at the variance and means across all samples to see, how bad is the situation:
```{r}
#calculating variances of different samples
var_column <- apply(bulk_time_course_ec[,-1], 2, var)

#calculating means of different samples
mean_column <- apply(bulk_time_course_ec[,-1], 2, mean)

#creating a data frame
df <- data.frame(Samples = names(var_column), Variance = var_column, Mean = mean_column)
rownames(df) <- c()

head(df)
```
```{r fig.width=10, fig.height=5,echo=FALSE}
variance <- ggplot(df, aes(x = Samples, y = Variance, group = 1, colour = "variance")) +
  geom_line(size=2) +
  theme_bw() +
  scale_color_manual(values=c("red")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

mean <- ggplot(df, aes(x = Samples, y = Mean, group = 1, colour = "mean")) +
  geom_line(size=2) +
  theme_bw() +
  scale_color_manual(values=c("blue")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

plot_grid(variance, mean, labels = "AUTO")
```

We can see that variance and means are both not the same at all for different samples (even the same replicates).
Let's look at H9_12_rep1 distribution of values:

```{r}
#removing gene column and transforming into matrix (for hist)
data <- as.matrix(bulk_time_course_ec[,-1])

hist(data, main="GSE75748", xlim = c(0,1500), xlab = "Expression",
     ylab = "Frequency", breaks = 300)
```

Let's try a log transformation here:

```{r}
hist(log2(data + 1), main="GSE75748 - log2 transformed", xlab = "Expression",
     ylab = "Frequency")
```

All datasets will include a mix of genes that are expressed and those that are not expressed. Whilst it is of interest
to examine genes that are expressed in one condition but not in another, some genes are unexpressed throughout all
samples. Let's check how many genes have zero expression across all 15 samples (in time series data):

```{r}
#changing the original data frame into log2
loged_bulk_time_course_ec <- bulk_time_course_ec
loged_bulk_time_course_ec[,2:dim(bulk_time_course_ec)[2]] <- log2(loged_bulk_time_course_ec[,
                                                                                        2:dim(bulk_time_course_ec)[2]]
                                                                  + 1)

table(rowSums(loged_bulk_time_course_ec[,-1]==0)==15)
```
We see that more than 15% of genes don't have any expression across all samples. Let's just delete them:
```{r}
#let's delete those genes that have less than 3 samples without zero expression
keep.exprs <- rowSums(loged_bulk_time_course_ec[,-1] > 0) > 3
cleaned_log_bulk_time_course_ec <- loged_bulk_time_course_ec[keep.exprs,]
```
```{r}
log_clean_data <- as.matrix(cleaned_log_bulk_time_course_ec[,-1])

hist(log_clean_data, main="cleaned GSE75748 - log2 transformed", xlab = "Expression",
     ylab = "Frequency")
```
Let's build boxplots and explore the data:

```{r}
meltedLogedBultTimeCourseEc <- melt(cleaned_log_bulk_time_course_ec, id='gene')

meltedLogedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

Looks much better now!

let's now perform RNA-seq analysis with limma, using only time factor variable (time column in metadata) and let's look separately at DE gene at each stage (from 12 to 24, from 24 to 36 and etc.):
```{r}
metadata_time <- metadata[,-3]
metadata_time
```
```{r}
designMatrix <- model.matrix(~0 + time, metadata_time)
head(designMatrix, 10) %>% kable()
```

```{r}
rownames(cleaned_log_bulk_time_course_ec) <- cleaned_log_bulk_time_course_ec$gene
cleaned_log_bulk_time_course_ec <- cleaned_log_bulk_time_course_ec[,-1]

# construct the contrast matrix
contrastMatrix <- makeContrasts(
  v24v12 = time24h - time12h,
  v36v24 = time36h - time24h,
  v72v36 = time72h - time36h,
  v96v72 = time96h - time72h,
  levels = designMatrix
)

contrastMatrix
```
```{r}
# keep the fit around as we will need to it for looking at other contrasts later 
time_course_Fit <- lmFit(cleaned_log_bulk_time_course_ec, designMatrix)

# fit the contrast using the original fitted model
contrastFit <- contrasts.fit(time_course_Fit, contrastMatrix)

# apply eBayes() for moderated statistics
contrastFitEb <- eBayes(contrastFit)

contrastGenes <- topTable(contrastFitEb)

contrastGenes
```

```{r}
cutoff <- 1e-04
time_course_res <- decideTests(contrastFitEb, p.value = cutoff, method = "global")
summary(time_course_res)
```

We see there are different number of genes up and down regulated at each stage. 

Here are the 34 genes that upregulated from 24 hours to 36 hours.

```{r}
hits1 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(v36v24 > 0)

hits1
```

```{r}
plotGenes <- function(genes, expressionMatrix, samplesMetadata) {
  
  expressionDataForGenes <- expressionMatrix %>% 
    rownames_to_column("gene") %>% 
    filter(gene %in% genes) %>%
    transformGeneExpressionMatrix() %>% 
    left_join(samplesMetadata, id = "sample_id")
  
  expressionDataForGenes %>% 
    ggplot(aes(x = dev_stage, y = expression, color = genotype)) +
    geom_point() +
    geom_jitter() +
    stat_summary(aes(y = expression, group=1), fun.y = mean, geom="line") +
    facet_wrap(~gene)
}
```

