---
title: "GSE75748 data analysis"
author: "German Novakovskiy"
date: '4 March 2018'
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Here we analyze gene expression data from [this paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1033-x). Some analysis steps are made according to [this workflow article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/).

```{r}
library(knitr)
suppressWarnings(suppressMessages(library(tidyverse)))
suppressWarnings(suppressMessages(library(GEOquery)))
suppressWarnings(suppressMessages(library(data.table)))
suppressWarnings(suppressMessages(library(reshape2)))
suppressWarnings(suppressMessages(library(dplyr)))
suppressWarnings(suppressMessages(library(ggplot2)))
suppressWarnings(suppressMessages(library(cowplot)))
suppressWarnings(suppressMessages(library(limma)))
suppressWarnings(suppressMessages(library(tibble)))
suppressWarnings(suppressMessages(library(RColorBrewer)))
suppressWarnings(suppressMessages(library(edgeR)))
suppressWarnings(suppressMessages(library(readxl)))
suppressWarnings(suppressMessages(library(VennDiagram)))
```

Let's read our data into R (GSE75748):
```{r}
if (file.exists("GSE75748.Rdata")) {
    # if previously downloaded
    load("GSE75748.Rdata")
} else {
    # Get geo object that contains our data and phenotype information
    geo_obj <- getGEO("GSE75748", GSEMatrix = TRUE)
    #geo_obj contains 1810 entries ((1018 + 758) single cells + (19 + 15) bulk RNA-seq)
    geo_obj <- geo_obj[[1]]
    save(geo_obj, file = "GSE75748.Rdata")
}
```

Further analysis of the downloaded data:
```{r}
#show data structure
show(geo_obj)

```
Let's load metadata:
```{r}
geo_metadata <- pData(geo_obj)

#due to complexity of metadata table it's more convenient to create your own (see below)
```

assayData has 0 features, it's not possible to load expression data from this dataset. That's why we will use expression data downloaded directly from the website
```{r}
#cell type
bulk_cell_type_ec <- read.csv("GSE75748_bulk_cell_type_ec.csv/GSE75748_bulk_cell_type_ec.csv", header=TRUE)
#time course
bulk_time_course_ec <- read.csv("GSE75748_bulk_time_course_ec.csv/GSE75748_bulk_time_course_ec.csv", header=TRUE)
```

Let's create metadata data frame:
```{r}
sample_names <- colnames(bulk_time_course_ec[-1]) #without gene name
time_fact <- paste(c(12,24,36,72,96), sep="", 'h') #factors of time
time_fact <- rep(time_fact, each=3)

time_int <- rep(c(12,24,36,72,96), each=3) #time as continious variable

metadata <- data.frame(samples = sample_names, time = time_fact, age = time_int)

metadata %>% kable()
```
```{r}
str(metadata)
```

Our primary goal here is to analyze bulk_time_course data (each stage with the next stage, step-by-step).

Our data here is in gene expected counts that were calculated using RSEM 1.2.3 (according to GEO query). Thus, we first have to transform them to CPM (counts per million) using edgeR (and also perform a library size normalization):

```{r}
#first let's create a edgeR DGElist object
bulk_time_course_ec <- as.matrix(bulk_time_course_ec)
rows <- bulk_time_course_ec[,1]
bulk_time_course_ec <- bulk_time_course_ec[,-1]
bulk_time_course_ec <- apply(bulk_time_course_ec, 2, as.double)
rownames(bulk_time_course_ec) <- rows

DGE_bulk_time_course_ec <- DGEList(counts = bulk_time_course_ec) 
normalized_factors_expression <- calcNormFactors(DGE_bulk_time_course_ec, method = "TMM") #calculation of scaling factors (for library size)

normalized_factors_expression$samples$norm.factors
```

For this dataset the effect of TMM-normalisation is mild, as evident in the magnitude of the scaling factors, which are all relatively close to 1.

Now let's calculate the cpm values for this expression dataset (taking into consideration the calculated above factors)
```{r}
cpm_bulk_time_course_expression <- cpm(normalized_factors_expression, log = FALSE)

head(cpm_bulk_time_course_expression) %>% kable()
```


Now let's do some sanity check: let's see how gene expression is distributed across different samples in different time points:

```{r}
#change first column name to gene
cpm_df <- as.data.frame(cpm_bulk_time_course_expression)
cpm_df <- cpm_df %>%
  rownames_to_column("gene")

meltedBultTimeCourseEc <- melt(cpm_df, id='gene')

meltedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```
Let's look at density plots:
```{r}
meltedBultTimeCourseEc %>% 
  ggplot(aes(x = value, color = variable)) +
  geom_density() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

We can look at the variance and means across all samples to see, how bad is the situation:
```{r}
#calculating variances of different samples
var_column <- apply(cpm_df[,-1], 2, var)

#calculating means of different samples
mean_column <- apply(cpm_df[,-1], 2, mean)

#creating a data frame
df <- data.frame(Samples = names(var_column), Variance = var_column, Mean = mean_column)
rownames(df) <- c()

head(df)
```
```{r fig.width=10, fig.height=5,echo=FALSE}
variance <- ggplot(df, aes(x = Samples, y = Variance, group = 1, colour = "variance")) +
  geom_line(size=2) +
  theme_bw() +
  scale_color_manual(values=c("red")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

mean <- ggplot(df, aes(x = Samples, y = Mean, group = 1, colour = "mean")) +
  geom_line(size=2) +
  theme_bw() +
  scale_color_manual(values=c("blue")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

plot_grid(variance, mean, labels = "AUTO")
```

We can see that variance and means are both not the same at all for different samples (especially for 72 hours samples).
Let's look at distribution of values:

```{r}
#removing gene column and transforming into matrix (for hist)
data <- as.matrix(cpm_df[,-1])

hist(data, main="GSE75748", xlim = c(0,1500), xlab = "Expression",
     ylab = "Frequency", breaks = 300)
```
Let's try a log transformation here:

```{r}
hist(log2(data + 0.25), main="GSE75748 - log2 transformed", xlab = "Expression",
     ylab = "Frequency")
```

All datasets will include a mix of genes that are expressed and those that are not expressed. Whilst it is of interest
to examine genes that are expressed in one condition but not in another, some genes are unexpressed throughout all
samples. Let's check how many genes have zero expression across all 15 samples (in time series data):

```{r}
#changing the original data frame into log2
log_cpm_df <- cpm_df
log_cpm_df[,2 : ncol(cpm_df)] <- log2(log_cpm_df[,2:ncol(cpm_df)] + 1)

table(rowSums(log_cpm_df[,-1]==0)==15)
```
We see that more than 18% of genes don't have any expression across all samples. Let's just delete them:
```{r}
#let's delete those genes that have less than 3 samples without zero expression
keep.exprs <- rowSums(log_cpm_df[,-1] > 0) > 3
cleaned_log_cpm_df <- log_cpm_df[keep.exprs,]
```

The number of remained genes:
```{r}
nrow(cleaned_log_cpm_df)
```

```{r}
log_clean_data <- as.matrix(cleaned_log_cpm_df[,-1])

hist(log_clean_data, main="cleaned GSE75748 - log2 transformed", xlab = "Expression",
     ylab = "Frequency")
```
Let's build boxplots and explore the data:

```{r}
meltedLogedBultTimeCourseEc <- melt(cleaned_log_cpm_df, id='gene')

meltedLogedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

Looks much better now!
```{r}
plotMDS(cleaned_log_cpm_df[,-1], cex=1.5)
```

It's clear that time ponts 12h, 24h and 36h are well separated from each other and 72h with 96h, however the last group (72 and 96) is clustered closer together, thus we don't expect to see much DE genes in these groups.

Let's now perform RNA-seq analysis with limma, using only time factor variable (time column in metadata) and let's look separately at DE gene at each stage (from 12 to 24, from 24 to 36 and etc.):
```{r}
metadata_time <- metadata[,-3]
metadata_time %>% kable()
```
```{r}
designMatrix <- model.matrix(~0 + time, metadata_time)
head(designMatrix, 10) %>% kable()
```

```{r}
rownames(cleaned_log_cpm_df) <- cleaned_log_cpm_df$gene
cleaned_log_cpm_df <- cleaned_log_cpm_df[,-1]

# construct the contrast matrix
contrastMatrix <- makeContrasts(
  v24v12 = time24h - time12h,
  v36v24 = time36h - time24h,
  v72v36 = time72h - time36h,
  v96v72 = time96h - time72h,
  levels = designMatrix
)

contrastMatrix %>% kable()
```
```{r}
# keep the fit around as we will need to it for looking at other contrasts later 
time_course_Fit <- lmFit(cleaned_log_cpm_df, designMatrix)

# fit the contrast using the original fitted model
contrastFit <- contrasts.fit(time_course_Fit, contrastMatrix)

# apply eBayes() for moderated statistics
contrastFitEb <- eBayes(contrastFit)

contrastGenes <- topTable(contrastFitEb)

contrastGenes %>% kable()
```

```{r}
cutoff <- 5e-02 #0.05 p value
#adjust method by default is BH (equivalent to fdr)
time_course_res <- decideTests(contrastFitEb, p.value = cutoff)
summary(time_course_res)
```

We see there are different number of genes up and down regulated at each stage. 

Here are the genes that upregulated from 24 hours to 36 hours.

```{r}
hits2 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(v36v24 > 0)

head(hits2) %>% kable()
```

```{r}
#function for plotting genes
plotGenes <- function(genes, expressionMatrix, metadata) {
  
  expressionDataForGenes <- expressionMatrix %>%
    rownames_to_column("gene") %>%
    filter(gene %in% genes) %>%
    melt()
  
  colnames(expressionDataForGenes) <- c("gene", "samples", "expression")
  expressionDataForGenes <- expressionDataForGenes %>%
    left_join(metadata, id="time")
  
  expressionDataForGenes %>% 
    ggplot(aes(x = time, y = expression, color=gene)) +
    geom_point() +
    geom_jitter() +
    stat_summary(aes(y = expression, group=1), fun.y = "mean", geom="line", size=2) +
    facet_wrap(~gene)
}

```

Let's plot 4 random genes that are upregulated at each stage.

```{r}
#stage 24v12
hits1 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(v24v12 > 0)

#stage 72v36
hits3 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(v72v36 > 0)

#stage 96v72
hits4 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(v96v72 > 0)
```
```{r}
sample_genes_24v12 <- sample(hits1$gene,4)
plotGenes(sample_genes_24v12, cleaned_log_cpm_df, metadata)
```
```{r}
sample_genes_36v24 <- sample(hits2$gene,4)
plotGenes(sample_genes_36v24, cleaned_log_cpm_df, metadata)
```

```{r}
sample_genes_72v36 <- sample(hits3$gene,4)
plotGenes(sample_genes_72v36, cleaned_log_cpm_df, metadata)
```
```{r}
sample_genes_96v72 <- sample(hits4$gene,4)
plotGenes(sample_genes_96v72, cleaned_log_cpm_df, metadata)
```

let's build those genes that according to the paper are DE in 24v12 transition:
```{r}
sample_genes <- c("T", "CDX1", "MSX2")
plotGenes(sample_genes, cleaned_log_cpm_df, metadata)
```

* COMMENTS *
- paper reports that CDX1, MSX2 and T are over expressed from 12 to 24h transition. I see only T and CDX1 (p-value cutoff 5e-02, FDR)
- paper reports that CER1 and GATA4 are over expressed from 24 to 36h transition.I see both (p-value cutoff 5e-02, FDR)
- paper reports that DKK4 and MYCT1 are over expressed from 36 to 72. I see both (p-value cutoff 5e-02, FDR)

Let's look at the expression of these genes EOMES, CER1, GATA4, PRDM1, and POU2AF1 at 96h (and KLF8 - "KLF8 may play a specific role during the transition from mesendoderm toward DE cells"):

```{r}
sample_genes <- c("EOMES", "CER1", "GATA4", "PRDM1", "POU2AF1", "KLF8")
plotGenes(sample_genes, cleaned_log_cpm_df, metadata)
```
Let's look at the expression of pluripotency genes POU5F1, NANOG, and SOX2:
```{r}
sample_genes <- c("POU5F1", "NANOG", "SOX2")
plotGenes(sample_genes, cleaned_log_cpm_df, metadata)
```
Their expression is going down, just as expected.

Now let's look at key DE markers CXCR4, SOX17, HNF1B, KIT, and KRT19:
```{r}
sample_genes <- c("CXCR4", "SOX17", "HNF1B", "KIT", "KRT19")
plotGenes(sample_genes, cleaned_log_cpm_df, metadata)
```
It's clear that CXCR4, SOX17 and KIT are upregulated at 96h, expression of KRT19 and HNF1B did not change so much, but KRT19 remained highly expressed, while HNF1B remained to be down-regulated.

# Comparisons with the paper 

"A total of 3247 differentially expressed genes were identified in the scRNA-seq time course experiment listed in Additional file 4: Table S3."

In my work I found this number of DE genes:
```{r}
#list all DE genes
allDEGenes <- topTable(contrastFitEb, number = Inf, p.value = 0.05)
nrow(allDEGenes)
```
According to Decide tests we have this number of DE genes (at different time stages):

```{r}
de_genes_rows <- apply(time_course_res, 1, function(x) any(x != 0))
de_genes_at_stages <- time_course_res[de_genes_rows,]
nrow(de_genes_at_stages)
```

We will check both numbers.
```{r}
all_de_genes <- rownames(allDEGenes)
stages_de_genes <- rownames(de_genes_at_stages)
```

Get list of DE genes from the study:
```{r}
paper_de_genes <- read_excel("13059_2016_1033_MOESM4_ESM.xlsx")
head(paper_de_genes) %>% kable()
```
```{r}
paper_de_genes <- paper_de_genes$GeneID
length(paper_de_genes)
```
So there is in fact 3247 genes in the result of this study.

Comparison of DE genes from decideTest with DE genes from the paper:


Comparison of DE genes from topTable with DE genes from the paper:
```{r}
temp <- venn.diagram(list(My_TopTable = all_de_genes, Paper_genes = paper_de_genes),fill = c("red", "green"), alpha = c(0.5, 0.5), cex = 2, cat.fontface = 4, lty =2, fontfamily =3, filename = NULL, main = "Comparison of paper DE genes with my all DE genes (from TopTable)", category.names = c("My topTable", "Paper genes"))

grid::grid.newpage()
grid::grid.draw(temp)
```

```{r}
temp2 <- venn.diagram(list(My_TopTable = stages_de_genes, Paper_genes = paper_de_genes),fill = c("red", "green"), alpha = c(0.5, 0.5), cex = 2, cat.fontface = 4, lty =2, fontfamily =3, filename = NULL, main = "Comparison of paper DE genes with my stage DE genes (from decideTests)", category.names = c("Stage DE genes", "Paper genes"))

grid::grid.newpage()
grid::grid.draw(temp2)
```


