---
title: "GSE75748 data analysis"
author: "German Novakovskiy"
date: '4 March 2018'
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Here we analyze gene expression data from [this paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1033-x). Some analysis steps are made according to [this workflow article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/).

```{r}
#source("https://bioconductor.org/biocLite.R")
#biocLite("GEOquery", "biomaRt")
library(GEOquery)
library(biomaRt)
#install.packages("tidyverse")
library(tidyverse)
#install.packages("data.table")
library(data.table)
#install.packages("reshape2")
library(reshape2)
library(dplyr)
library(ggplot2)
library(cowplot)
```

Let's read our data into R (GSE75748):
```{r}
if (file.exists("GSE75748.Rdata")) {
    # if previously downloaded
    load("GSE75748.Rdata")
} else {
    # Get geo object that contains our data and phenotype information
    geo_obj <- getGEO("GSE75748", GSEMatrix = TRUE)
    #geo_obj contains 1810 entries ((1018 + 758) single cells + (19 + 15) bulk RNA-seq)
    geo_obj <- geo_obj[[1]]
    save(geo_obj, file = "GSE75748.Rdata")
}
```

Further analysis of the downloaded data:
```{r}
#show data structure
show(geo_obj)
```

assayData has 0 features, it's not possible to load expression data from this dataset. That's why we will use expression data downloaded directly from the website
```{r}
#cell type
bulk_cell_type_ec <- read.csv("GSE75748_bulk_cell_type_ec.csv/GSE75748_bulk_cell_type_ec.csv", header=TRUE)
#time course
bulk_time_course_ec <- read.csv("GSE75748_bulk_time_course_ec.csv/GSE75748_bulk_time_course_ec.csv", header=TRUE)
```

Our primary goal here is to analyze bulk_time_course data (each stage with the next stage, step-by-step).

Now let's do some sanity check: let's see how gene expression is distributed across different samples in different time points:

```{r}
#change first column name to gene
colnames(bulk_time_course_ec)[1] <- "gene"
meltedBultTimeCourseEc <- melt(bulk_time_course_ec, id='gene')

meltedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```
Let's look at density plots:
```{r}
meltedBultTimeCourseEc %>% 
  ggplot(aes(x = value, color = variable)) +
  geom_density() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

We can look at the variance and means across all samples to see, how bad is the situation:
```{r}
#calculating variances of different samples
var_column <- apply(bulk_time_course_ec[,-1], 2, var)

#calculating means of different samples
mean_column <- apply(bulk_time_course_ec[,-1], 2, mean)

#creating a data frame
df <- data.frame(Samples = names(var_column), Variance = var_column, Mean = mean_column)
rownames(df) <- c()

head(df)
```
```{r fig.width=10, fig.height=5,echo=FALSE}
variance <- ggplot(df, aes(x = Samples, y = Variance, group = 1, colour = "variance")) +
  geom_line(size=2) +
  theme_bw() +
  scale_color_manual(values=c("red")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

mean <- ggplot(df, aes(x = Samples, y = Mean, group = 1, colour = "mean")) +
  geom_line(size=2) +
  theme_bw() +
  scale_color_manual(values=c("blue")) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

plot_grid(variance, mean, labels = "AUTO")
```

We can see that variance and means are both not the same at all for different samples (even the same replicates).
Let's look at H9_12_rep1 distribution of values:

```{r}
#removing gene column and transforming into matrix (for hist)
data <- as.matrix(bulk_time_course_ec[,-1])

hist(data, main="GSE75748", xlim = c(0,1500), xlab = "Expression",
     ylab = "Frequency", breaks = 300)
```

Let's try a log transformation here:

```{r}

hist(log2(data + 1), main="GSE75748 - log2 transformed", xlab = "Expression",
     ylab = "Frequency")
```

```{r}
#changing the original data frame into log2
loged_bulk_time_course_ec <- bulk_time_course_ec
loged_bulk_time_course_ec[,2:dim(bulk_time_course_ec)[2]] <- log2(loged_bulk_time_course_ec[,
                                                                                        2:dim(bulk_time_course_ec)[2]]
                                                                  + 1)

meltedBultTimeCourseEc <- melt(bulk_time_course_ec, id='gene')

meltedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```


