---
title: "GSE98411 data analysis"
author: "German Novakovskiy"
date: "August 23, 2018"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r, echo=FALSE}
library(knitr)
suppressWarnings(suppressMessages(library(tidyverse)))
suppressWarnings(suppressMessages(library(GEOquery)))
suppressWarnings(suppressMessages(library(data.table)))
suppressWarnings(suppressMessages(library(reshape2)))
suppressWarnings(suppressMessages(library(dplyr)))
suppressWarnings(suppressMessages(library(ggplot2)))
suppressWarnings(suppressMessages(library(cowplot)))
suppressWarnings(suppressMessages(library(limma)))
suppressWarnings(suppressMessages(library(tibble)))
suppressWarnings(suppressMessages(library(RColorBrewer)))
suppressWarnings(suppressMessages(library(edgeR)))
suppressWarnings(suppressMessages(library(readxl)))
suppressWarnings(suppressMessages(library(VennDiagram)))
suppressMessages(suppressWarnings(library(ermineR)))
suppressMessages(suppressWarnings(library(hgu133a.db)))
suppressMessages(suppressWarnings(library(stringr)))
suppressMessages(suppressWarnings(library(DESeq2)))
suppressMessages(suppressWarnings(library(clusterProfiler)))
suppressMessages(suppressWarnings(library(biomaRt)))
```

Here we analyze gene expression data from [this data set.](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE98411).\

Differentiation protocol: day 1 (Primitive streak induction) media included 100 ng/mL Activin A, 50 nM PI-103 (PI3K inhibitor), 2 nM CHIR99021 (Wnt agonist), days 2-3 media included 100 ng/mL Activin A and 250 nM LDN-193189 (BMP inhibitor). 

```{r}
if (file.exists("GSE98411.Rdata")) {
    # if previously downloaded
    load("GSE98411.Rdata")
} else {
    # Get geo object that contains our data and phenotype information
    geo_obj <- getGEO("GSE98411", GSEMatrix = TRUE)
    geo_obj <- geo_obj[[1]]
    save(geo_obj, file = "GSE98411.Rdata")
}
```
```{r}
show(geo_obj)
```

Constructing an appropriate metadata matrix:
```{r}
#get covariate matrix
geo_metadata <- pData(geo_obj)[, c("organism_ch1", "title", colnames(pData(geo_obj))[grep("characteristics", 
    colnames(pData(geo_obj)))])]

#splitting the column with cell line and age info
x <- str_split_fixed(geo_metadata$title, "_", 2)

#geo_metadata <- geo_metadata[,-2] will be used for filtering raw counts
geo_metadata$cell_type <- x[,1]
geo_metadata$age <- x[,2]

#delete columns characteristics_ch1 and characteristics_ch1.2
geo_metadata <- geo_metadata[,c(-3,-5)]

#splitting the sex column
x <- str_split_fixed(geo_metadata[,3], ": ", 2)
geo_metadata <- geo_metadata[,-3]
geo_metadata$sex <- x[,2]

colnames(geo_metadata) <- c("organism", "title", "cell", "age", "sex")

geo_metadata 
```

Collapsing info in geo_metadata:
```{r}
old_samples <- geo_metadata$title
#tech_replicates rows
tech_rep <- c(1,2,6,7,9,10,14,15,17,18,22,23,25,26,30,31)

geo_metadata <- geo_metadata[-tech_rep,]
geo_metadata$title <- as.character(geo_metadata$title)
rownames(geo_metadata) <- NULL

geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H1_D0", "H1", "D0", "Female"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H5_D0", "H5", "D0", "Male"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H1_D1", "H1", "D1", "Female"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H5_D1", "H5", "D1", "Male"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H1_D2", "H1", "D2", "Female"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H5_D2", "H5", "D2", "Male"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H1_D3", "H1", "D3", "Female"))
geo_metadata <- rbind(geo_metadata, c("Homo sapiens", "H5_D3", "H5", "D3", "Male"))

geo_metadata <- geo_metadata[c(17,1,2,3,18,4,19,5,6,7,20,8,21,9,10,11,22,12,23,13,14,15,24,16),]
rownames(geo_metadata) <- NULL

geo_metadata$title <- as.factor(geo_metadata$title)
geo_metadata$age <- as.factor(geo_metadata$age)
geo_metadata$sex <- as.factor(geo_metadata$sex)
geo_metadata$cell <- as.factor(geo_metadata$cell)

geo_metadata
```

Let's load the expression data (in raw counts - after featureCounts):
```{r}
expr_data_counts <- read.table("GSE98411_RNA_counts.txt", header=TRUE) 

expr_data_counts <- expr_data_counts[, colnames(expr_data_counts) %in% old_samples]

#colnames(expr_data_counts)[-1] <- as.character(geo_metadata$sample)

head(expr_data_counts, 10) #%>% kable()
```
```{r}
dim(expr_data_counts)
```

Usually one will filter genes with 10-15 read counts. We have r`min(DGE_bulk_time_course_ec$samples$lib.size)` as a lowest library size. 10 reads in such library size will be: r`10/min(DGE_bulk_time_course_ec$samples$lib.size)*1e6`. It is aproximately 2 cpm (the filtering is to this threshold) But 2 is to strict, let's go with 1.5
```{r}
##first let's create a edgeR DGElist object

rownams <- rownames(expr_data_counts)

expr_data_counts <- as.matrix(expr_data_counts)
exp_data_counts_matrix <- apply(expr_data_counts, 2, function(x) as.numeric(x))
rownames(exp_data_counts_matrix) <- rownams

DGE_bulk_time_course_ec <- DGEList(counts = exp_data_counts_matrix) 

cpm <- cpm(DGE_bulk_time_course_ec)
keep.exprs <-rowSums(cpm > 1.5) >= 8 #8 samples for each condition including technical replicates

DGE_bulk_time_course_ec <- DGE_bulk_time_course_ec[keep.exprs,,]

dim(DGE_bulk_time_course_ec)
```
```{r}
#converting to SYMBOL ID
x <- rownames(DGE_bulk_time_course_ec$counts)

#biomart really gives pretty the same result as bitr
#open connection between biomaRt and R. 
#human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
#names <- getBM( attributes=c("ensembl_gene_id", "hgnc_symbol") , filters= "ensembl_gene_id", values = unique(x), mart = human)

eg <-  bitr(x, fromType="ENSEMBL", toType="SYMBOL", OrgDb="org.Hs.eg.db")
```
```{r}
dim(eg)
```
```{r}
length(unique(eg$ENSEMBL))
```
```{r}
length(unique(eg$SYMBOL))
```
```{r}
#delete replicates
eg <- eg[-which(duplicated(eg$ENSEMBL)),]
eg <- eg[-which(duplicated(eg$SYMBOL)),]

new_df <- DGE_bulk_time_course_ec$counts %>%
  as.data.frame() %>%
  rownames_to_column("ENSEMBL") %>%
  filter(ENSEMBL %in% eg$ENSEMBL) %>%
  merge(eg) %>%
  column_to_rownames("SYMBOL")

new_df <- new_df[,-1]
new_df <- as.matrix(new_df)

DGE_bulk_time_course_ec <- DGEList(counts = new_df) 
```


We should collapse technical replicates:
```{r}
temp_df <- DGE_bulk_time_course_ec$counts

H1_D0 <- temp_df[,1] + temp_df[,2]
H5_D0 <- temp_df[,6] + temp_df[,7] 
H1_D1 <- temp_df[,9] + temp_df[,10]
H5_D1 <- temp_df[,14] + temp_df[,15] 
H1_D2 <- temp_df[,17] + temp_df[,18]
H5_D2 <- temp_df[,22] + temp_df[,23] 
H1_D3 <- temp_df[,25] + temp_df[,26]
H5_D3 <- temp_df[,30] + temp_df[,31] 

temp_df <- temp_df[,-tech_rep]

#assigning new columns (merged replicates)
temp_df <- cbind(temp_df, H1_D0)
temp_df <- cbind(temp_df, H5_D0)
temp_df <- cbind(temp_df, H1_D1)
temp_df <- cbind(temp_df, H5_D1)
temp_df <- cbind(temp_df, H1_D2)
temp_df <- cbind(temp_df, H5_D2)
temp_df <- cbind(temp_df, H1_D3)
temp_df <- cbind(temp_df, H5_D3)

#stupid, I know, but I need things fast
temp_df <- temp_df[,c(17,1,2,3,18,4,19,5,6,7,20,8,21,9,10,11,22,12,23,13,14,15,24,16)]
DGE_bulk_time_course_ec <- DGEList(counts = temp_df) 

head(DGE_bulk_time_course_ec$counts)
```

We have raw counts, thus we need to perform a library size normalization using edgeR:
```{r}
normalized_factors_expression <- calcNormFactors(DGE_bulk_time_course_ec, method = "TMM") #calculation of scaling factors (for library size)

normalized_factors_expression$samples$norm.factors
```

Let's look at distribution of values:
```{r}
#removing gene column and transforming into matrix (for hist)
data <- as.matrix(DGE_bulk_time_course_ec$counts)

hist(data, main="GSE98411", xlim = c(0,200000), xlab = "Expression",
     ylab = "Frequency", breaks = 300)
```

Let's now perform RNA-seq analysis with limma, using age, sex, cell columns and let's look separately at DE gene at each stage:
```{r}
metadata_age <- geo_metadata[,-1]
metadata_age
```

First, let's use contrast matrix:
```{r}
designMatrix <- model.matrix(~0 + age + sex, metadata_age)
head(designMatrix, 10) 
```

We can apply voom (which usually takes count data as an input and transforms them to logCPM) that estimates the mean-variance relationship and uses this to compute appropriate observation-level weights. The data are then ready for linear modelling.

"Moreover,  the  voom-plot  provides  a  visual  check  on  the  level  of  filtering  performed  upstream.  If  filtering  of
lowly-expressed  genes  is  insufficient,  a  drop  in  variance  levels  can  be  observed  at  the  low  end  of  the  expression scale  due  to  very  small  counts.  If  this  is  observed,  one  should  return  to  the  earlier  filtering  step  and  increase  the expression threshold applied to the dataset."
```{r}
after_voom_cpm <- voom(normalized_factors_expression, designMatrix, plot=TRUE)
```

```{r}
hist(after_voom_cpm$E, main="cleaned GSE98411 - log2 transformed CPM", xlab = "Expression",
     ylab = "Frequency")
```

Boxplots to explore the data:
```{r}
cleaned_log_cpm_df <- as.data.frame(after_voom_cpm$E)

cleaned_log_cpm_df <- cleaned_log_cpm_df %>% rownames_to_column("gene")

meltedLogedBultTimeCourseEc <- melt(cleaned_log_cpm_df, id='gene')

meltedLogedBultTimeCourseEc %>%
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```
```{r}
plotMDS(cleaned_log_cpm_df[,-1], cex=1.5)
```

Let's create a contrast matrix because we are interested in DE genes across different time points:
```{r}
rownames(cleaned_log_cpm_df) <- cleaned_log_cpm_df$gene
cleaned_log_cpm_df <- cleaned_log_cpm_df[,-1]

# construct the contrast matrix
contrastMatrix <- makeContrasts(
  D1vD0 = ageD1 - ageD0,
  D2vD1 = ageD2 - ageD1,
  D3vD2 = ageD3 - ageD2,
  D3vD1 = ageD3 - ageD1,
  D3vD0 = ageD3 - ageD0,
  levels = designMatrix
)

#contrastMatrix #%>% kable()
```

```{r}
# keep the fit around as we will need to it for looking at other contrasts later 
time_course_Fit <- lmFit(after_voom_cpm, designMatrix)

# fit the contrast using the original fitted model
contrastFit <- contrasts.fit(time_course_Fit, contrastMatrix)

# apply eBayes() for moderated statistics
contrastFitEb <- eBayes(contrastFit)

#contrastGenes <- topTable(contrastFitEb, number = Inf, p.value = 0.05)

plotSA(contrastFitEb)
```
```{r}
cutoff <- 5e-02 #0.05 p value
#adjust method by default is BH (equivalent to fdr)
time_course_res <- decideTests(contrastFitEb, p.value = cutoff, lfc = 1)
summary(time_course_res)
```

Let's look at top upregulated genes at different stages:
```{r}
hits1 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(D1vD0 > 0)


hits2 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(D2vD1 > 0)


hits3 <- time_course_res %>% 
  as.data.frame() %>% 
  rownames_to_column("gene") %>% 
  filter(D3vD2 > 0)
```

```{r}
#function for plotting genes
plotGenes <- function(genes, expressionMatrix, metadata) {
  
  expressionDataForGenes <- expressionMatrix %>%
    rownames_to_column("gene") %>%
    filter(gene %in% genes) %>%
    melt()
  
  colnames(expressionDataForGenes) <- c("gene", "title", "expression")
  expressionDataForGenes <- expressionDataForGenes %>%
    left_join(metadata, id="age")
  
  expressionDataForGenes %>% 
    ggplot(aes(x = age, y = expression, color=gene)) +
    geom_point() +
    geom_jitter() +
    stat_summary(aes(y = expression, group=1), fun.y = "mean", geom="line", size=2) +
    facet_wrap(~gene)
}
```

Comparisons to key genes from previous analysis.

```{r message=FALSE, warning=FALSE}
#CDX1 - ENSG00000113722
#SOX17 - ENSG00000164736
sample_genes <- c("CDX1", "SOX17", "GATA6")
plotGenes(sample_genes, cleaned_log_cpm_df, metadata_age)
```
```{r message=FALSE, warning=FALSE}
#SOX2 - ENSG00000181449
#FOXA2 - ENSG00000125798
sample_genes <- c("FOXA2", "EOMES", "T")
plotGenes(sample_genes, cleaned_log_cpm_df, metadata_age)
```

For pathway analysis
```{r}
#we have only 3 days
DEgenes_0h_72h <- topTable(contrastFitEb, number = Inf, coef = "D3vD0")
DEgenes_0h_24h <- topTable(contrastFitEb, number = Inf, coef = "D1vD0")
DEgenes_24h_72h <- topTable(contrastFitEb, number = Inf, coef = "D3vD1")

#for pathway enrichment analysis
DEgenes_0h_72h_98411 <- DEgenes_0h_72h
DEgenes_0h_24h_98411 <- DEgenes_0h_24h
DEgenes_24h_72h_98411 <- DEgenes_24h_72h
save(DEgenes_0h_72h_98411, file = "DEgenes_0h_72h_98411.Rdata")
save(DEgenes_0h_24h_98411, file = "DEgenes_0h_24h_98411.Rdata")
save(DEgenes_24h_72h_98411, file = "DEgenes_24h_72h_98411.Rdata")
################################

```

## Gene set enrichment analysis of GSE98411

```{r}
if (!file.exists("GO.xml")) { goToday("GO.xml") }

ggplot(data = DEgenes_0h_72h, aes(x = logFC, y = -log(adj.P.Val), color = (-log(adj.P.Val) > 3)))+
  scale_colour_manual(name = 'p-value < 0.05', values = setNames(c('red','black'),c(T, F)), labels = c("False", "True"))+
  geom_point()+
  geom_vline(xintercept=0)+
  geom_vline(xintercept=-2)+
  geom_vline(xintercept=2)+
  #xlim(-1.5,1.5)+
  ylab("-log(p-value)")+
  xlab("logFC")+
  labs(title="Gene expression differences in 0h and 72h cells, GSE98411")+
  theme_bw()+
  theme(axis.line = element_line(colour = "black"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        #panel.border = element_blank(),
        panel.background = element_blank()) +
  theme(axis.title.x=element_text(size=12),
        axis.text.y=element_text(size=12),
        axis.title.y=element_text(size=14),
        axis.ticks.x=element_blank(),
        strip.text.x = element_text(size=14),
        strip.background = element_rect(colour="white", fill="white"),
        legend.text=element_text(size=15),
        legend.title=element_text(size=14))
```
```{r}
ermineInputGeneScores <- DEgenes_0h_72h %>% 
  rownames_to_column("gene") %>%
  #mutate(absolute_logFC = abs(logFC)) %>% 
  dplyr::select(gene, logFC) %>% 
  na.omit() %>% 
  as.data.frame() %>% 
  arrange(desc(logFC)) %>% 
  column_to_rownames("gene")

head(ermineInputGeneScores, 10) %>% kable() # print the first few rows
```

```{r}
enrichmentResult <- precRecall(scores = ermineInputGeneScores, 
                               scoreColumn = 1, # column 1 is the scores 
                               bigIsBetter = TRUE, # larger logFC should be ranked higher
                               annotation = "Generic_human", # ask ermineJ to use the Generic_human annotation file (will automatically download)
                               aspects = "B", # look at only biological processes 
                               iterations = 10000, # 10K sampling iterations so that results are stable
                               geneSetDescription = "GO.xml") # use the GO XML file in current directory

enrichmentResult$results %>% arrange(MFPvalue) %>% head(10) %>% kable()
```

```{r}
enrichmentResult$results %>% 
  dplyr::select(Name, CorrectedPvalue, CorrectedMFPvalue) %>% 
  arrange(CorrectedMFPvalue) %>% 
  head(10) %>% 
  kable(align = "l", col.names = c("Biological Process", "Corrected p-value", 
                                   "Corrected MF p-value"))
```
```{r fig.width=15, fig.height=10}
Enrichment <- enrichmentResult$results
Enrichment$Name <- as.factor(Enrichment$Name)

Enrichment %>% 
  dplyr::select(Name, NumGenes, CorrectedMFPvalue) %>% 
  arrange(CorrectedMFPvalue) %>% 
  filter(CorrectedMFPvalue <= 5e-2) %>% 
  head(25) %>% 
  ggplot(aes(x = fct_reorder(Name, CorrectedMFPvalue), 
             y = NumGenes, fill = CorrectedMFPvalue)) +
  geom_col() +
  labs(title = "Biological Processes - GSE98411", 
       x = "", y = "Number of Genes") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  coord_flip() +
  theme_light() 

```

